OpenSim/MenuManager.swift

```diff
@@ -169,5 +210,47 @@ protocol MenuManagerDelegate {
         sender.state = (wasOn ? .off : .on)
         setLaunchAtLogin(itemUrl: Bundle.main.bundleURL, enabled: !wasOn)
     }
+
+    private func eraseAllSimulators(omitBootedSimulators: Bool) {
```

Comment:
I think we can improve the readability of this by removing the flag. Methods with flags are a bit harder to read and maintain, my suggestion:
Delete this `eraseAllSimulators` function, and add 3 new functions:
```
private func resetAllSimulators() { }
    
private func resetShutdownSimulators() { }

private func resetSimulators(_ devices: [Device]) { }
```

`resetSimulators` is a generic function that receives an array of devices and resets them, it's agnostic to state or condition, it just resets whatever it receives. `resetShutdownSimulators` and `resetAllSimulators` will just filter the simulators you want and then send them to `resetSimulators`

By doing this `resetSimulators` function doesn't need conditional checking and will look cleaner, all the logic to get the correct simulators is defined on `resetShutdownSimulators` and `resetAllSimulators` which gives each function a single goal.

That way we have a clear understanding of what each method does, and it will make the callstack easier to read
from: 
`eraseAllSimulators(omitBootedSimulators: false)` 
to
`resetAllSimulators` 

and 
`eraseAllSimulators(omitBootedSimulators: true)`
to
`resetShutdownSimulators`

Reply:
Thank you for this suggestion, I agree that implementation made it a bit unclear. I've modified as per your suggestion.